applications/luci-app-tailscale/htdocs/luci-static/resources/view/tailscale/setting.js

```diff
@@ -0,0 +1,265 @@
+/* SPDX-License-Identifier: GPL-3.0-only
+ *
+ * Copyright (C) 2024 asvow
+ */
+
+'use strict';
+'require form';
+'require fs';
+'require network';
+'require poll';
+'require rpc';
+'require uci';
+'require view';
+
+var callServiceList = rpc.declare({
+	object: 'service',
+	method: 'list',
+	params: ['name'],
+	expect: { '': {} }
+});
+
+function getInterfaceSubnets(interfaces = ['lan', 'wan']) {
+    return network.getNetworks().then(networks => {
+        return [...new Set(
+            networks
+               .filter(ifc => interfaces.includes(ifc.getName()))
+               .flatMap(ifc => ifc.getIPAddrs())
+               .filter(addr => addr.includes('/'))
+               .map(addr => {
+                    const [ip, cidr] = addr.split('/');
+                    const ipParts = ip.split('.').map(Number);
+                    const mask = ~((1 << (32 - parseInt(cidr))) - 1);
+                    const subnetParts = ipParts.map((part, i) => (part & (mask >> (24 - i * 8))) & 255);
+                    return `${subnetParts.join('.')}/${cidr}`;
+                })
+        )];
+    });
+}
+
+function getStatus() {
+	var status = {
+		isRunning: false,
+		backendState: undefined,
+		authURL: undefined,
+		displayName: undefined,
+		onlineExitNodes: [],
+		subnetRoutes: []
+	};
+	return Promise.resolve(callServiceList('tailscale')).then(res => {
```

Comment:
`callServiceList` should already return a promise, so no need to wrap it in one.
```suggestion
	const res = await callServiceList('tailscale');
```